File: Project/MyProject/MyProject/Source/MyProject/Private/BasicFrame/Libs/FrameHandle/TickMgr.cpp
#include "MyProject.h"
#include "TickMgr.h"
#include "ITickedObject.h"
#include "TickProcessObject.h"
#include "MClassFactory.h"
#include "IDelayHandleItem.h"
#include "MacroDef.h"
#include "Ctx.h"
#include "LogSys.h"
#include "LogTypeId.h"

MY_BEGIN_NAMESPACE(MyNS)

M_IMPLEMENT_AND_REGISTER_CLASS(TickMgr, TickObjectPriorityMgr)

TickMgr::TickMgr()
{

}

TickMgr::~TickMgr()
{

}

void TickMgr::init()
{
	Super::init();
}

void TickMgr::dispose()
{
	Super::dispose();
}

void TickMgr::addTick(ITickedObject* tickObj, float priority)
{
	this->_addObject((IDelayHandleItem*)tickObj, priority);
}

void TickMgr::removeTick(ITickedObject* tickObj)
{
	this->_removeObject((IDelayHandleItem*)tickObj);
}

void TickMgr::advance(float delta, TickMode tickMode)
{
	Super::advance(delta, tickMode);
}

MY_END_NAMESPACE

File: Project/MyProject/Tool/CSToolKit/Project/ToolKit/ToolKit/Source/Libs/FrameHandle/TickMgr.cs
namespace SDK.Lib
{
    /**
     * @brief 心跳管理器
     */
    public class TickMgr : TickObjectPriorityMgr
    {
        public TickMgr()
        {
            
        }

        override public void init()
        {
            base.init();
        }

        override public void dispose()
        {
            base.dispose();
        }

        override public void advance(float delta, TickMode tickMode)
        {
            if (MacroDef.ENABLE_PROFILE)
            {
                Ctx.msInstance.mProfiler.enter("TickMgr::Advance");
            }

            base.advance(delta, tickMode);

            if (MacroDef.ENABLE_PROFILE)
            {
                Ctx.msInstance.mProfiler.exit("TickMgr::Advance");
            }
        }

        public void addTick(ITickedObject tickObj, float priority = 0.0f)
        {
            this._addObject(tickObj as IDelayHandleItem, priority);
        }

        public void removeTick(ITickedObject tickObj)
        {
            this._removeObject(tickObj as IDelayHandleItem);
        }
    }
}